plugins/outputs/clarify/clarify.go

```diff
@@ -0,0 +1,154 @@
+package clarify
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+	"github.com/clarify/clarify-go/views"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Clarify struct {
+	Username        string          `toml:"username"`
+	Password        string          `toml:"password"`
+	CredentialsFile string          `toml:"credentials_file"`
+	IDTags          []string        `toml:"id_tags"`
+	Log             telegraf.Logger `toml:"-"`
+
+	client *clarify.Client
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (c *Clarify) Connect() error {
+	ctx := context.Background()
+	if c.CredentialsFile != "" {
+		creds, err := clarify.CredentialsFromFile(c.CredentialsFile)
+		if err != nil {
+			return err
+		}
+		c.client = creds.Client(ctx)
+		return nil
+	}
+	if c.Username != "" && c.Password != "" {
+		creds := clarify.BasicAuthCredentials(c.Username, c.Password)
+		c.client = creds.Client(ctx)
+		return nil
+	}
+	return fmt.Errorf("no Clarify credentials provided")
+}
+
+func verifyValue(v interface{}) (float64, error) {
+	var value float64
+	switch v := v.(type) {
+	case bool:
+		value = float64(0)
+		if v {
+			value = float64(1)
+		}
+	case uint8:
+		value = float64(v)
+	case uint16:
+		value = float64(v)
+	case uint32:
+		value = float64(v)
+	case uint64:
+		value = float64(v)
+	case int8:
+		value = float64(v)
+	case int16:
+		value = float64(v)
+	case int32:
+		value = float64(v)
+	case int64:
+		value = float64(v)
+	case float32:
+		value = float64(v)
+	case float64:
+		value = v
+	default:
+		return value, fmt.Errorf("unsupported field type: %T", v)
+	}
+	return value, nil
+}
+
+func (c *Clarify) Write(metrics []telegraf.Metric) error {
+	signals := make(map[string]views.SignalSave)
+	frame := views.DataFrame{}
+
+	for _, m := range metrics {
+		for _, f := range m.FieldList() {
+			if value, err := verifyValue(f.Value); err == nil {
+				id := c.generateID(m, f)
+				ts := fields.AsTimestamp(m.Time())
+
+				if _, ok := frame[id]; ok {
+					frame[id][ts] = value
+				} else {
+					frame[id] = views.DataSeries{ts: value}
+				}
+
+				s := views.SignalSave{}
+				s.Name = fmt.Sprintf("%s.%s", m.Name(), f.Key)
```

Comment:
No, we want to keep the names as human readable as possible. The ID is added to the signal along with all the tags as labels, so it will be easy to find it and sort it in Clarify.